restcomm/restcomm.dao/src/main/java/org/restcomm/connect/dao/mybatis/MybatisIncomingPhoneNumbersDao.java

```diff
@@ -70,19 +74,61 @@ public IncomingPhoneNumber getIncomingPhoneNumber(final String phoneNumber) {
         return getIncomingPhoneNumber("getIncomingPhoneNumberByValue", phoneNumber);
     }
 
-    private IncomingPhoneNumber getIncomingPhoneNumber(final String selector, Object parameter) {
+   private IncomingPhoneNumber getIncomingPhoneNumber(final String selector, Object parameter) {
         final SqlSession session = sessions.openSession();
-        try {
+        String inboundPhoneNumber = null;
+          try {
+            //if(parameter.toString().contains("*")){}{
+            //perform direct check
             final Map<String, Object> result = session.selectOne(namespace + selector, parameter);
-            if (result != null) {
+            if (result != null ) {
                 return toIncomingPhoneNumber(result);
-            } else {
-                return null;
-            }
-        } finally {
+            }//}
+            //check if there is a Regex match only if parameter is a String aka phone Number
+            if(!(parameter instanceof Sid)){
+               inboundPhoneNumber = parameter.toString().replace("+1", "");
```

Comment:
We should remove the https://github.com/RestComm/Restcomm-Connect/blob/e1f429084ef147ba4b2e1ae78a65e683c037a9e4/restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java#L953-L953 line that adds the +1 to all number that are not formated with internation access code. 
@deruelle please confirm if you agree